axelor-business-project/src/main/java/com/axelor/apps/businessproject/db/repo/ProjectBusinessProjectRepository.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2021 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.businessproject.db.repo;
+
+import com.axelor.apps.base.db.AppBusinessProject;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.repo.SequenceRepository;
+import com.axelor.apps.base.service.administration.SequenceService;
+import com.axelor.apps.businessproject.exception.IExceptionMessage;
+import com.axelor.apps.businessproject.service.app.AppBusinessProjectService;
+import com.axelor.apps.hr.db.repo.ProjectHRRepository;
+import com.axelor.apps.project.db.Project;
+import com.axelor.common.StringUtils;
+import com.axelor.exception.AxelorException;
+import com.axelor.exception.db.repo.TraceBackRepository;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import javax.persistence.PersistenceException;
+
+public class ProjectBusinessProjectRepository extends ProjectHRRepository {
+
+  @Override
+  public Project save(Project project) {
+    try {
+      AppBusinessProject appBusinessProject =
+          Beans.get(AppBusinessProjectService.class).getAppBusinessProject();
+
+      if (StringUtils.isBlank(project.getCode())
+          && appBusinessProject.getGenerateProjectSequence()) {
+        Company company = project.getCompany();
+        String seq =
+            Beans.get(SequenceService.class)
+                .getSequenceNumber(SequenceRepository.PROJECT_SEQUENCE, company);
+
+        if (seq == null) {
+          throw new AxelorException(
+              company,
+              TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+              I18n.get(IExceptionMessage.PROJECT_SEQUENCE_ERROR),
+              company.getName());
+        }
+
+        project.setCode(seq);
+      }
+    } catch (AxelorException e) {
+      throw new PersistenceException(e.getLocalizedMessage());
```

Comment:
Please change this line to
```suggestion
      throw new PersistenceException(e);
```
to avoid losing the stacktrace when an exception occurs. On top of that we can also trace the AxelorException before rethrowing it so we are sure to trace it.